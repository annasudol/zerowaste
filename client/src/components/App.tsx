import * as React from "react";
import { Navigation } from "../components";
import { Login, SignUp, Main, RecipesList, AddRecipeForm, RecipeInfo, UserInfo } from "../pages";

import { AppRoutes } from '../../routes';
import { Route, Switch } from 'react-router-dom';
import { ErrorMessage } from './ErrorMessage';


export const App: React.FunctionComponent = () => {
    const token = window.localStorage.getItem('token');

    return (
        <>
            <Navigation token={token} />
            <Switch>
                <Route path={AppRoutes.Home} component={Main} exact />
                <Route path={AppRoutes.RecipesList} component={RecipesList} />
                <Route path={AppRoutes.Recipe} component={RecipeInfo} />

                {!token ? (
                    <>
                        <Route path={AppRoutes.Login} component={Login} />
                        <Route path={AppRoutes.SignUp} component={SignUp} />
                    </>
                ) : (
                        <>
                            <Route path={AppRoutes.User} component={UserInfo} />
                            <Route path={AppRoutes.AddRecipe} component={AddRecipeForm} />
                        </>
                    )}
                <Route path="*">
                    <ErrorMessage message="page not found" />
                </Route>
            </Switch>
        </>
    )
}