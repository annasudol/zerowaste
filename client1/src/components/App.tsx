import * as React from "react";



// tslint:disable-next-line: one-variable-per-declaration
export const App: React.FunctionComponent = () => {

    return (<div>
        <p className="font-gotham">hello</p>
    </div>)
}