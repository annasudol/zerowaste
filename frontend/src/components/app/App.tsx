import React, { FC, ReactElement } from 'react';
import { Navigation, PublicRoute, PrivateRoute, ErrorMessage } from '../../components';
import { Login, SignUp, Home, RecipesList, AddRecipeForm, RecipeInfo, UserInfo, EditRecipeForm } from '../../pages';
import { AppRoutes } from '../../routes'
;
import { Route, Switch } from 'react-router-dom';


export const App: FC = (): ReactElement => (
        <>
        <div className="left"></div>
        <div className="right"></div>
            <Navigation />
            <Switch>
                <Route exact path={AppRoutes.Home} component={Home} />
                <PublicRoute path={AppRoutes.Login} component={Login} />
                <PublicRoute path={AppRoutes.SignUp} component={SignUp} />
                <PublicRoute path={AppRoutes.RecipesList} component={RecipesList} />
                <PublicRoute path={AppRoutes.Recipe} component={RecipeInfo} />
                <PrivateRoute path={AppRoutes.AddRecipe} component={AddRecipeForm} />
                <PrivateRoute path={AppRoutes.EditRecipe} component={EditRecipeForm} />
                <PrivateRoute path={AppRoutes.User} component={UserInfo} />
                <Route component={ErrorMessage} />
            </Switch>
        </>
    )
