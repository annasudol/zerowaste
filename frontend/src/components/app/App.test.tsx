import { shallow } from 'enzyme';
import React from 'react';

import { App } from './App';
import toJson from 'enzyme-to-json';

describe('component', (): void => {
  describe('App', (): void => {
    it('match snapshot', (): void => {
      const wrapper = shallow(<App/>,);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
