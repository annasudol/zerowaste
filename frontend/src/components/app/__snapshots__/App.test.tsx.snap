// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component App match snapshot 1`] = `
<div>
  <div
    className="left"
  />
  <div
    className="right"
  />
  <Navigation />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <PublicRoute
      component={[Function]}
      path="/login"
    />
    <PublicRoute
      component={[Function]}
      path="/signUp"
    />
    <PublicRoute
      component={[Function]}
      path="/recipes"
    />
    <PublicRoute
      component={[Function]}
      path="/recipe/:recipeID"
    />
    <PrivateRoute
      component={[Function]}
      path="/addRecipe"
    />
    <PrivateRoute
      component={[Function]}
      path="/editRecipe/:recipeID"
    />
    <PrivateRoute
      component={[Function]}
      path="/user"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
